### PR TITLE
Introduce minimal API layer for drawing on Canvas (using WebGL bindings)

### PR DESCRIPTION
**Pull request by Schahen**

### CODE REVIEW

File: skiko/src/jsMain/kotlin/org/jetbrains/skia/impl/Native.js.kt [hunk 1]
```diff
@@ -21,7 +21,7 @@ actual abstract class Native actual constructor(ptr: NativePointer) {
 }
 
 actual fun reachabilityBarrier(obj: Any?) {
-    TODO()
+    // TODO: impl later
```
Comment (olonho) [inline skiko/src/jsMain/kotlin/org/jetbrains/skia/impl/Native.js.kt:24]: Guess it's just not needed.

File: skiko/src/jsMain/kotlin/org/jetbrains/skiko/wasm/Wrapper.kt [hunk 2]
```diff
@@ -2,12 +2,79 @@ package org.jetbrains.skiko.wasm
...
+    val preferLowPowerToHighPerformance: Int?
+    val failIfMajorPerformanceCaveat: Int?
+    val enableExtensionsByDefault: Int?
+    val explicitSwapControl: Int?
+    val renderViaOffscreenBackBuffer: Int?
+    val majorVersion: Int?
+}
+
+fun ContextAttributes.asJsObject(): dynamic {
+    val jsObject = js("{}")
```
Comment (olonho) [inline skiko/src/jsMain/kotlin/org/jetbrains/skiko/wasm/Wrapper.kt:40]: Wouldn't 'val jsObject: dynamic = Any()' be better?

File: skiko/src/jsMain/kotlin/org/jetbrains/skiko/wasm/Wrapper.kt [hunk 3]
```diff
@@ -2,12 +2,79 @@ package org.jetbrains.skiko.wasm
...
+        override val majorVersion = attr?.majorVersion ?: 1
+    }
+
+    val contextPointer = GL.createContext(canvas, contextAttributes.asJsObject())
+    return GL.makeContextCurrent(contextPointer)
+}
+
+
 @JsExport
 fun createSkikoCanvas(): HTMLCanvasElement {
```
Comment (olonho) [inline skiko/src/jsMain/kotlin/org/jetbrains/skiko/wasm/Wrapper.kt:79]: Maybe remove this function?

File: skiko/src/jsMain/kotlin/org/jetbrains/skiko/wasm/Wrapper.kt (hunk 2 above)
Comment (Schahen) [reply skiko/src/jsMain/kotlin/org/jetbrains/skiko/wasm/Wrapper.kt:40]: any and dynamic are different beasts in Kotlin/JS - Any is a standard "regular" type and still lacks possibility to refer to arbitrary fields (like dynamic does)

File: skiko/src/jsTest/kotlin/WasmTests.kt [hunk 4]
```diff
@@ -26,6 +23,5 @@ private fun wasmTest(block: () -> Unit) = GlobalScope.promise {
 class WasmTests {
     @Test
     fun pingTest() = wasmTest {
```
Comment (olonho) [inline skiko/src/jsTest/kotlin/WasmTests.kt:25]: Maybe remove ping test as well?

File: skiko/src/jsMain/kotlin/org/jetbrains/skiko/wasm/Wrapper.kt (hunk 2 above)
Comment (olonho) [reply skiko/src/jsMain/kotlin/org/jetbrains/skiko/wasm/Wrapper.kt:40]: OK, just looks somewhat unusual.

**Review by olonho — APPROVED** [review]

**Pull request merged**